src/argilla/utils/modeling.py

```diff
@@ -0,0 +1,241 @@
+        bool_cols = df.select_dtypes(include=["boolean"]).columns
+        float_cols = df.select_dtypes(include=["float64"]).columns
+        # Cast integer columns to Python's native int type
+        df[int_cols] = df[int_cols].astype(int)
+        # Cast boolean columns to Python's native str type
+        df[bool_cols] = df[bool_cols].astype(str)
+        # Cast float columns to Python's native float type and round to 2 decimal places
+        df[float_cols] = df[float_cols].astype(float).round(2)
+        return df
+
+    def _create_metadata_properties(self, df: pd.DataFrame) -> List:
+        """
+        Generate metadata properties based on dataframe columns and data types.
+
+        Args:
+            df (pd.DataFrame): The text descriptives dataframe.
+
+        Returns:
+            List: A list of metadata properties.
+        """
+        properties = []
+        for col, dtype in df.dtypes.items():
+            name = col
+            title = name.replace("_", " ").title()
+            if dtype in ["object", "bool"]:
+                prop = TermsMetadataProperty(name=name, title=title, visible_for_annotators=self.visible_for_annotators, values=["True", "False"])
+            elif dtype == "int64":
+                prop = IntegerMetadataProperty(
+                    name=name, title=title, visible_for_annotators=self.visible_for_annotators
+                )
+            elif dtype == "float64":
+                prop = FloatMetadataProperty(name=name, title=title, visible_for_annotators=self.visible_for_annotators)
+            else:
+                print(f"Unhandled data type for column {col}: {dtype}")
+                continue
+            properties.append(prop)
+        return properties
+
+    def _add_text_descriptives_to_metadata(
+        self, records: List[Union[FeedbackRecord, RemoteFeedbackRecord]], df: pd.DataFrame
+    ) -> List[Union[FeedbackRecord, RemoteFeedbackRecord]]:
+        """
+        Add the text descriptives metrics extracted previously as metadata
+        to a list of FeedbackDataset records.
+
+        Args:
+            records (List[Union[FeedbackRecord, RemoteFeedbackRecord]]): A list of FeedbackDataset or RemoteFeedbackDataset records.
+            df (pd.DataFrame): The text descriptives dataframe.
+
+        Returns:
+            List[Union[FeedbackRecord, RemoteFeedbackRecord]]: A list of FeedbackDataset or RemoteFeedbackDataset records with extracted metrics added as metadata.
+        """
+        # Loop through the records and add the metrics to the metadata
+        modified_records = []
+        # Make tqdm optional
+        for record, metrics in tqdm(
+            zip(records, df.to_dict("records")),
+            total=len(records),
+            disable=not self.progress,
+            desc="Getting text descriptives...",
+        ):
+            record.metadata.update(metrics)
+            modified_records.append(record)
+        return modified_records
+
+    def update_dataset(
```

Comment:
I think we might want the option to use the methods separately too. It might be that I already have a dataset with metadata properties, and we just want to update the records. It could also be that we only want to use a limited set of records to avoid duplicating the computation, it would be difficult to do that now.

Reply:
I added a method for `.update_records()`